Library/Coupon.cs

```diff
@@ -71,7 +71,7 @@ public enum CouponType
         public CouponDiscountType DiscountType { get; private set; }
         public CouponState State { get; private set; }
         public RedemptionResourceType RedemptionResource { get; set; }
-        public CouponType Type { get; set; } = CouponType.SingleCode;
```

Comment:
@bhelx I didn't realize that is C# 6.0 syntax for setting the default value for a property. It didn't compile for me and I thought I had accidentally put it there so I removed it. My change should be reverted.